sharding/shardset.go

```diff
@@ -93,9 +93,9 @@ func validateShards(shards []uint32) error {
 	return nil
 }
 
-// DefaultHashGen generates a HashFn based on murmur32
+// DefaultHashGen generates a HashFn based on ID's internal hash.
 func DefaultHashGen(length int) HashFn {
-	return func(id string) uint32 {
-		return murmur3.Sum32([]byte(id)) % uint32(length)
+	return func(id ts.ID) uint32 {
+		return uint32(id.Hash().Fold() % uint64(length))
```

Comment:
Don't be worried, one of the points of hash functions is to provide a uniform distribution of the output regardless of the input, which is both applicable to map keys and shard assignment.